server/src/main/java/io/crate/expression/operator/any/AnyEqOperator.java

```diff
@@ -61,14 +62,18 @@ boolean matches(Object probe, Object candidate) {
     protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, @NotNull List<?> nonNullValues, Context context) {
         String columnName = probe.storageIdent();
         DataType<?> innerType = ArrayType.unnest(probe.valueType());
+        if (nonNullValues.get(0) instanceof Collection<?>) {
+            // nested_array_ref = any([[1], [1,2]])
+            return termsAndGenericFilter(any, probe, nonNullValues, context);
+        }
```

Comment:
I'd tend to check `ArrayType.dimensions` on `probe.valueType()` instead of accessing the first value to not rely on the parent never passing empty arrays.